cloud/pkg/edgecontroller/controller/upstream.go

```diff
@@ -1004,6 +1008,12 @@ func NewUpstreamController(config *v1alpha1.EdgeController, factory k8sinformer.
 	uc.podLister = factory.Core().V1().Pods().Lister()
 	uc.configMapLister = factory.Core().V1().ConfigMaps().Lister()
 	uc.secretLister = factory.Core().V1().Secrets().Lister()
+	factory.Core().V1().PersistentVolumes()
+	uc.pvLister = factory.Core().V1().PersistentVolumes().Lister()
+	factory.Core().V1().PersistentVolumeClaims()
+	uc.pvcLister = factory.Core().V1().PersistentVolumeClaims().Lister()
+	factory.Storage().V1().VolumeAttachments()
+	uc.attachLister = factory.Storage().V1().VolumeAttachments().Lister()
```

Comment:
It seems that you just need to call `factory...Lister()` directly.

Reply:
+1

Reply:
+1